FSD light noise update

cuddandr opened the pull request:
Includes updates to the yamls to use a new light noise file for the FSD simulation.

This also includes a couple changes to `simulate_pixels.py` to account for inaccuracies in floating point math when calculating `digit_samples`, and a change to the logic statement for loading the light LUT.

The new calculation of `digit_samples` has been checked using the 2x2 and ND-LAr values to give the same result.

wforeman committed:
Move segment t0 calculation and delayted-t0 filtering earlier before segment_ids are defined

wforeman committed:
Remove some obsolete commented-out code

wforeman committed:
Add some warnings when segments are dropped

cuddandr committed:
Update FSD configs to use new light noise and light window

cuddandr committed:
Further adjust light windows for FSD

cuddandr committed:
Adjust digit_samples calculation to fix inaccuracy

digit_samples uses the ceil() function to get an integer, however this
can produce an unexpected result if the floating point calculation
produces a value slightly higher than it should assuming infinite
precision.

For example, 0.96+8.64 should equal 9.6 exactly, however it evalulates
to 9.600000000000001 when running. This error can cause ceil() to then
return a higher integer than expected. To complete the example for the
FSD parameters, this results in waveforms with 601 instead of 600 ticks.

This fix uses round() to reduce the calculation of the light_trig_window
size to three decimal places before using ceil() to eliminate the
inaccuracy at very high precision.

The use of round() inside the ceil() function should retain the correct
behavior if the light_trig_window is not an even multiple of the
digit_sample_spacing

krwood commented on larndsim/detector_properties/fsd.yaml:
```diff
@@ -32,11 +32,12 @@ tau_t: 1.530 # us
-light_window: [0, 16] # us
-light_trig_window: [1.6, 14.4] # us
+light_window: [0.00, 8.64] # us
+light_trig_window: [0.96, 8.64] # us
```
I _think_ this should be

`light_trig_window: [0.96, 7.68] # us`

But should be double checked with an expert...

cuddandr replied:
@AWh1t3 and I worked out the `light_trig_window` (and `light_window`) to get the correctly sized FSD waveforms, but I'll have to ask again why this is the correct window (namely the end point being the same as `light_window`).

cuddandr committed:
Enable HDF5 dataset compression

Added an option to turn on compression of the HDF5 datasets as a command
line argument to simulate_pixels.py

By default it is off, and compresses all datasets except for 'packets'
as that dataset is created by the larpix package and not in larndsim.

Commit:
Channel-by-channel thresholds in FSD

cuddandr committed:
Set 'light_trig_mode' to 0 for FSD simulation

Empirically produces more sensible results for the charge+light
simluation for the FSD.

This could be revisited as we produce more simulation and make changes.

cuddandr committed:
Merge branch 'develop' into fsd_cosmic_sim

Fixed conflict in cli/simulate_pixels.py where the fsd_cosmic_sim branch
added a round() function to prevent an inaccuracy due to floating point
precision when calculating digit_samples

Commit:
Update thresholds with the new gain value

Commit:
Update the gain value

Commit:
Update default threshold value

cuddandr committed:
Merge pull request #296 from DUNE/fsd_cosmic_sim_gain_updates

FSD pixel thresholds updated for simulation and updated the default gain for the FSD detector properties from Jafaar.
